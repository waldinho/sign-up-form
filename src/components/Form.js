import React, { useState } from 'react';
import styled from 'styled-components';

const SubscribeForm = () => {
    
    const [firstname, setFirstname] = useState()
    const [surname, setSurname] = useState()
    const [email, setEmail] = useState()
    const [phone, setPhone] = useState()
    const [submission, setSubmission] = useState()
    const [message, setMessage] = useState()


    const handleSubmit = e => {
        e.preventDefault()
        const url ='https://ckzvgrbymezqegu.form.io/reacttestform/submission'
        const data = { 
            data: {
                firstName: firstname, 
                lastName: surname,
                email: email,
                mobilePhone: phone,
            }
        }
        fetch(url, { 
            method: 'POST',
            body: JSON.stringify(data),
            headers:{ 
                'Content-Type': 'application/json',
                'x-auth': 'react-test' 
            } 
        })
        .then(res => res.json())
        .catch(error => console.error('Error:', error),
            setSubmission(false),
            setMessage('Oops something went wrong...'),
        )
        .then(response => console.log('Success:', response),
            setSubmission(true),
            setMessage('Thank you for subscribing!'),
        )
    }

    return (
        <Wrapper>
            {submission 
            ? 
            <div className='thank__you'>
                <p>{message}</p>
            </div>
            :
            <form onSubmit={handleSubmit}>
                <div className='row'>
                    <div className='left'>
                        <label>First name: *</label>
                        <input 
                            type="text" 
                            name="firstname" 
                            onChange={e => {setFirstname(e.target.value)}}
                            required
                        />
                    </div>
                    <div className='right'>
                        <label>Last name: *</label>
                        <input 
                            type="text" 
                            name="surnamname" 
                            onChange={e => {setSurname(e.target.value)}}
                            required
                        />
                    </div>
                </div>
                <div className='row'>
                    <div className='left'>
                        <label>Email: *</label>
                        <input 
                            type="email" 
                            name="email" 
                            onChange={e => {setEmail(e.target.value)}} 
                            required
                        />
                    </div>
                    <div className='right'>
                        <label>Mobile phone: </label>
                        <input 
                            type="tel"
                            name="phone" 
                            onChange={e => {setPhone(e.target.value)}}
                            maxLength="13"
                            minLength="10"
                        />
                    </div>
                </div>
                <input 
                    className='submit'
                    type="submit" 
                    value="Submit" 
                /> 
            </form>
            }
        </Wrapper>
    )
}
    
const Wrapper = styled.div`
    position: relative;
    top: 26vh;
    @media screen and (min-width: 600px) {
        top: 30vh;
    }
    color: #fff;
    max-width: 1024px;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    text-align: left;
    .thank__you {
        position: relative;
        top: 10vh;
    }
    form {
        display: flex;
        flex-direction: column;
        .row {
            display: flex;
            flex-direction: column;
            @media screen and (min-width: 600px) {
                flex-direction: row;
            }
            .left {
                display: flex;
                flex-direction: column;
                justify-content: flex-start;
            }
            .right {
                display: flex;
                flex-direction: column;
                justify-content: flex-end;
            }
            input {
                border: 2px solid #dadada;
                box-sizing: border-box;
                color: #323232;
                font-size: 18px;
                height: 48px;
                line-height: 25.5px;
                padding: 10.5px 14px 14px 14px;
                font-family: 'Muli', sans-serif;
                margin: 1rem 0.75rem 0 1rem;
                @media screen and (min-width: 321px) {
                    margin: 1rem;
                }
                min-width: 300px;
            }
            label {
                min-width: 300px;
                padding: 10.5px 14px 0 14px;
            }
        }
        input.submit {
            background: #333;
            color: #fff;
            border: 2px solid #dadada;
            box-sizing: border-box;
            font-size: 18px;
            height: 48px;
            line-height: 25.5px;
            font-family: 'Muli', sans-serif;
            margin: 2.5rem 0.75rem 0 1rem;
            @media screen and (min-width: 321px) {
                margin: 2.5rem 1rem;
            }
            @media screen and (min-width: 600px) {
                width: 150px;
            }
        }
    }
`

export default SubscribeForm