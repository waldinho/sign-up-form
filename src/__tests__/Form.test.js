import React from 'react';
import ReactDOM from 'react-dom';
import Form from '../components/Form';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const handleSubmit = jest.fn().mockImplementation((cb) => () => cb({ test: 'test' }));
const wrapper = shallow(<Form form="test" handleSubmit={handleSubmit}   />);

const firstname = wrapper.find('input').at(0);
firstname.simulate('change', { target: { value: 'some text' } });

const surname = wrapper.find('input').at(1);
surname.simulate('change', { target: { value: 'some text' } });

const email = wrapper.find('input').at(2);
email.simulate('change', { target: { value: 'some text' } });

const phone = wrapper.find('input').at(3);
phone.simulate('change', { target: { value: 'some text' } });

describe('<Form />', function() {
    it('Should have <label> with First name label title', function(){
        const component = shallow(<Form />);
        expect(component.find('label').at(0).text()).toEqual('First name: *');
    });
    it('Should have <label> with Surname label title', function(){
        const component = shallow(<Form />);
        expect(component.find('label').at(1).text()).toEqual('Last name: *');
    });
    it('Should have <label> with Email label title', function(){
        const component = shallow(<Form />);
        expect(component.find('label').at(2).text()).toEqual('Email: *');
    });
    it('Should have <label> with Mobile phone label title', function(){
        const component = shallow(<Form />);
        expect(component.find('label').at(3).text()).toEqual('Mobile phone: ');
    });
})