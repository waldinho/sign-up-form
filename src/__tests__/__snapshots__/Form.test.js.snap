// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa hVPNmO\\"><form><div class=\\"row\\"><div class=\\"left\\"><label>First name: *</label><input type=\\"text\\" name=\\"firstname\\" required=\\"\\"/></div><div class=\\"right\\"><label>Last name: *</label><input type=\\"text\\" name=\\"surnamname\\" required=\\"\\"/></div></div><div class=\\"row\\"><div class=\\"left\\"><label>Email: *</label><input type=\\"email\\" name=\\"email\\" required=\\"\\"/></div><div class=\\"right\\"><label>Mobile phone: </label><input type=\\"tel\\" name=\\"phone\\" maxLength=\\"13\\" minLength=\\"10\\"/></div></div><input type=\\"submit\\" class=\\"submit\\" value=\\"Submit\\"/></form></div>"`;

exports[`<Form /> with no props should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa hVPNmO\\"><form><div class=\\"row\\"><div class=\\"left\\"><label>First name: *</label><input type=\\"text\\" name=\\"firstname\\" required=\\"\\"/></div><div class=\\"right\\"><label>Last name: *</label><input type=\\"text\\" name=\\"surnamname\\" required=\\"\\"/></div></div><div class=\\"row\\"><div class=\\"left\\"><label>Email: *</label><input type=\\"email\\" name=\\"email\\" required=\\"\\"/></div><div class=\\"right\\"><label>Mobile phone: </label><input type=\\"tel\\" name=\\"phone\\" maxLength=\\"13\\" minLength=\\"10\\"/></div></div><input type=\\"submit\\" class=\\"submit\\" value=\\"Submit\\"/></form></div>"`;
